modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/Extensions.kt

```diff
@@ -378,6 +382,13 @@ fun IdBox.extractId(): Any = when (this) {
     is IdBox.ParameterId -> this.parameterId
 }
 
+fun InstructionBox.extractAccount() = this
+    .cast<InstructionBox.Register>()
```

Comment:
we've got plenty of those, ofc a calling party must understand what they deal with, also I guess it's usually obvious about the target type, try to look for usages of these things

Reply:
anyway if there is even a preferable alternative there should be a task for the entire project to apply a redesign created, not within this PR

Reply:
> we've got plenty of those

ok, my 2 cents. IMO, the others look quite fine. For example every cast here is perfectly fine because the hierarchy is guaranteed by the schema (these casts here should never fail unless something was changed in `core`):
```kt
fun InstructionBox.Register.extractAccount() = this
    .registerBox.`object`.expression
    .cast<Expression.Raw>().value
    .cast<Value.Identifiable>().identifiableBox
    .cast<IdentifiableBox.NewAccount>().newAccount
``` 

But to downcast `InstructionBox` to `InstructionBox.Register` is quite possible to fail:
```kt
fun InstructionBox.extractAccount() = this
    .cast<InstructionBox.Register>()
```

Reply:
I'm not saying not to do the cast. I'd just not want to see it exposed as a high level API to have `extractAccount` on `InstructionBox`. The cast should be contained at the call site

Reply:
I see your point, didn't realize it was quite narrow initially, thanks for the explanation
@arndey please check

Reply:
we use this, pretty handy
if SDK client invoke this on wrong InstructionBox - he will get an exception, what's the problem?

Reply:
@mversic it seems your suggestion will just move the first cast from inside of the extension function to the client side which is basically the same in terms of the overall result but less elegant imo (imagine lots of client code with such casts, it will just complicate the 'business' side code with boilerplate)
we should rather consider an alternative complex approach if applicable within a separate activity